src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,36 @@
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(actor);
+            transaction.commit();
+        } catch (RuntimeException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException(
+                    "Can't save actor to DB : " + actor.getName(), e
+            );
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return actor;
     }
 
     @Override
     public Optional<Actor> get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            Actor actor = session.get(Actor.class, id);
+            session.close();
+            return Optional.ofNullable(actor);
+        } catch (RuntimeException e) {
+            throw new DataProcessingException("Can't get an actor with id : " + id);
```

Comment:
The `DataProcessingException` constructor call in the catch block of the `get` method is missing the cause exception `e`. Consider adding it to provide more context for the exception: `throw new DataProcessingException("Can't get an actor with id : " + id, e);`